src/api/TradeAPI.ts

```diff
@@ -113,4 +134,130 @@ export default class TradeAPI {
+   * @param  fromTokenDecimals    Token decimals of token being sold (ex: 18)
+   * @param  toTokenDecimals      Token decimals of token being bought (ex: 18)
+   * @param  rawAmount            String quantity of token to sell (ex: "0.5")
+   * @param  fromAddress          SetToken address which holds the buy / sell components
+   * @param  setToken             SetTokenAPI instance
+   * @param  gasPrice             (Optional) gasPrice to calculate gas costs with (Default: fetched from GasNow)
+   * @param  slippagePercentage   (Optional) maximum slippage, determines min receive quantity. (Default: 2%)
+   * @param  isFirmQuote          (Optional) Whether quote request is indicative or firm
+   * @param  feePercentage        (Optional) Default: 0
+   * @param  feeRecipient         (Optional) Default: 0xD3D555Bb655AcBA9452bfC6D7cEa8cC7b3628C55
+   * @param  excludedSources      (Optional) Exchanges to exclude (Default: ['Kyber', 'Eth2Dai', 'Uniswap', 'Mesh'])
+   *
+   * @return {Promise<TradeQuote>}
+   */
+  public async fetchTradeQuoteAsync(
+    fromToken: Address,
+    toToken: Address,
+    fromTokenDecimals: number,
+    toTokenDecimals: number,
+    rawAmount: string,
+    fromAddress: Address,
+    setToken: SetTokenAPI,
+    gasPrice?: number,
+    slippagePercentage?: number,
+    isFirmQuote?: boolean,
+    feePercentage?: number,
+    feeRecipient?: Address,
+    excludedSources?: string[],
+  ): Promise<TradeQuote> {
+    this.assert.schema.isValidAddress('fromToken', fromToken);
+    this.assert.schema.isValidAddress('toToken', toToken);
+    this.assert.schema.isValidAddress('fromAddress', fromAddress);
+    this.assert.schema.isValidJsNumber('fromTokenDecimals', fromTokenDecimals);
+    this.assert.schema.isValidJsNumber('toTokenDecimals', toTokenDecimals);
+    this.assert.schema.isValidString('rawAmount', rawAmount);
+
+    const chainId = (await this.provider.getNetwork()).chainId;
+
+    return this.tradeQuoter.generate({
+      fromToken,
+      toToken,
+      fromTokenDecimals,
+      toTokenDecimals,
+      rawAmount,
+      fromAddress,
+      chainId,
+      tradeModule: this.tradeModuleWrapper,
+      provider: this.provider,
+      setToken,
+      gasPrice,
+      slippagePercentage,
+      isFirmQuote,
+      feePercentage,
+      feeRecipient,
+      excludedSources,
+    });
+  }
+
+  /**
+   * Fetches a list of tokens and their metadata from CoinGecko. Each entry includes
+   * the token's address, proper name, decimals, exchange symbol and a logo URI if available.
+   * For Ethereum, this is a list of tokens tradeable on Uniswap, for Polygon it's a list of
+   * tokens tradeable on Sushiswap's Polygon exchange. Method is useful for acquiring token decimals
+   * necessary to generate a trade quote and images for representing available tokens in a UI.
+   *
+   * @return List of tradeable tokens for chain platform
+   */
+  public async fetchTokenListAsync(): Promise<CoinGeckoTokenData[]> {
```

Comment:
Just want to confirm this is not currently being consumed? It was part of the old flow for grabbing the decimals? This might belong in a different utils type of API

Reply:
We're using this stuff in the admin UI. This API is helpful I think.